e2e-tests/cypress/support/helpers/SmartEsClient.ts

```diff
@@ -0,0 +1,45 @@
+export class SmartEsClient {
+
+  static pruneAllReportingIndices() {
+    cy.log('Pruning all reporting indices...');
+    cy.getRequest({ url: `${Cypress.env().elasticsearchUrl}/_cat/indices?format=json`, user: 'kibana:kibana' })
+      .then((result: GetIndices[]) => {
+        result
+          .filter(index => index.index.startsWith('.reporting'))
+          .forEach(reportingIndex => {
+            SmartEsClient.deleteIndexDocsByQuery(reportingIndex.index);
+            SmartEsClient.refreshIndex(reportingIndex.index);
+          });
+      });
+    cy.log('Pruning all reporting indices - DONE!');
+  }
+
+  static deleteIndexDocsByQuery = (index: string) => {
+    SmartEsClient.callEsEndpointUsingKibanaUser(
+      "POST", `${index}/_delete_by_query`, `-H "Content-Type: application/json" -d '{"query": {"match_all": {}}}'`
+    )
+  }
+
+  static refreshIndex = (index: string) => {
+    SmartEsClient.callEsEndpointUsingKibanaUser("POST", `${index}/_refresh`)
+  }
+
+  static deleteIndex = (index: string) => {
+    SmartEsClient.callEsEndpointUsingKibanaUser("DELETE", index)
+  }
+
+  static addDocument = (index: string, id: string, doc: Record<string, any>) => {
+    SmartEsClient.callEsEndpointUsingKibanaUser("POST", `${index}/_doc/${id}`, `-H "Content-Type: application/json" -d '${JSON.stringify(doc)}'`)
+  }
+
+  static callEsEndpointUsingKibanaUser = (method: String, endpoint: String, curlParams: String = "") => {
```

Comment:
Do we need this static method? IMO we can use our custom Cypress [commands](https://github.com/beshu-tech/readonlyrest-e2e-tests/blob/a30392566fbd75d707db670cedc5916d93b4ce35/e2e-tests/cypress/support/commands.ts#L4-L3) like:
```
static kibanaUser = 'kibana:kibana'
...
cy.post({url: `${Cypress.env().elasticsearchUrl}/${index}/_refresh`, user: SmartEsClient.kibanaUser}) 
```

Reply:
ok, I removed the method. But at the same time I decided to refactor the post/put/import/get/delete commands. WDYT about my refactoring?

Reply:
Looks good, one thing where we could improve DX is this place `${Cypress.env().login}:${Cypress.env().password}`. We probably could move it to the static variable like `static public userCredentials = ${Cypress.env().login}:${Cypress.env().password}`